acme/client.go

```diff
@@ -670,6 +670,7 @@ func (c *Client) requestCertificateForCsr(authz []authorizationResource, bundle
 
 			break
 		default:
+			defer resp.Body.Close()
```

Comment:
Why not put this defer directly after the post, i.e. line 612? Then the `Close` in the other switch can be removed as well.